### PR TITLE
DAOS-11393 control: Log numa stats after hugepage cleanup

### PR DESCRIPTION
**Pull request by tanabarr**
Print output of numastat -m to the daos_admin log after hugepages have
been cleaned up which happens just before engine start. This provides
per-NUMA details of memory allocations and usage which can be used to
troubleshoot insufficient hugepage memory failures on engine start-up.

Required-githooks: true

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Review by kjacque — APPROVED** [review]

**Review by knard-intel — COMMENTED** [review]
LGFM :)

**Review by mjmac — APPROVED** [review]
Looks OK overall, just a couple of things to fix if this needs to be repushed. We just need to be sensitive to increasing log spam.

File: src/control/server/storage/bdev/backend.go [hunk 1]
```diff
@@ -172,6 +173,22 @@ func cleanHugePages(log logging.Logger, hugePageDir string) (count uint, _ error
...
+	if err != nil {
+		toLog = (&runCmdError{
+			wrapped: err,
+			stdout:  string(out),
+		}).Error()
+	} else {
+		toLog = string(out)
+	}
+
+	log.Debugf("run cmd numastat -m: %s", toLog)
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/storage/bdev/backend.go:189]: this is where the info is output, without it this function is meaningless

File: src/control/server/storage/bdev/backend.go [hunk 2]
```diff
@@ -172,6 +173,22 @@ func cleanHugePages(log logging.Logger, hugePageDir string) (count uint, _ error
...
 
+func logNUMAStats(log logging.Logger) {
+	var toLog string
+
+	out, err := exec.Command("numastat", "-m").Output()
+	if err != nil {
+		toLog = (&runCmdError{
+			wrapped: err,
+			stdout:  string(out),
+		}).Error()
```
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/control/server/storage/bdev/backend.go:184]: This isn't logging error, its recording the message in "toLog" variable to be printed to Debug on line 189.

**Review by knard-intel — APPROVED** [review]

**Review by mjmac — APPROVED** [review]

**Pull request merged**